integration_tests/main.rs

```diff
@@ -204,28 +350,68 @@ impl Test {
                 }),
             }?;
 
-            // Check customer status
-            let customer_detail_json = customer_cli!(Show, vec!["show", &self.name, "--json"])
-                .run(rng.clone(), customer_config.clone())
-                .await
-                .context("Failed to show customer channel")?;
-
-            let customer_channel: customer::zkchannels::PublicChannelDetails =
-                serde_json::from_str(&customer_detail_json)?;
-
-            assert_eq!(customer_channel.status(), expected_outcome.customer_status);
-
-            // Check merchant status
-            let channel_id = &customer_channel.channel_id().to_string();
-            let merchant_details_json = merchant_cli!(Show, vec!["show", channel_id, "--json"])
-                .run(merchant_config.clone())
-                .await
-                .context("Failed to show merchant channel")?;
-
-            let merchant_channel: merchant::zkchannels::PublicChannelDetails =
-                serde_json::from_str(&merchant_details_json)?;
-
-            assert_eq!(merchant_channel.status(), expected_outcome.merchant_status);
+            match expected_outcome.channel_outcome {
+                None => return Ok(()),
+                Some(ChannelOutcome {
+                    customer_status: expected_customer_status,
+                    merchant_status: expected_merchant_status,
+                    customer_balance: expected_customer_balance,
+                    merchant_balance: expected_merchant_balance,
+                }) => {
```

Comment:
suggestion: To reduce nesting here, you can get the channel outcome out of the match statement but do all the actual validation after the match.

```
let ChannelOutcome { ... } = match expected_outcome.channel_outcome {
    None => return Ok(()),
    Some(channel_outcome) => channel_outcome,
};

// rest of the function
```

Reply:
It seems doing this binds the expected_outcome variable and to resolve that, you need to match on `&expected_outcome.channel_outcome` instead, but then the ChannelOutcome variables become references. Tests still pass if you dereference every time you need to use the variables though, so is this still fine?

i.e. we're matching on `&expected_outcome.channel_outcome` so we have to keep referring to the ChannelOutcome members as `*expected_{member}`